### PR TITLE
DB-7875 support repeat() function for String

### PR DESCRIPTION
**Pull request by yxia92**
Please refer to [DB-7875](https://splicemachine.atlassian.net/browse/DB-7875) for details.

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj [hunk 1]
```diff
@@ -16897,6 +16911,7 @@ nonReservedKeyword()  :
 // SQL92 says it is reserved, but we want it to be non-reserved.
 	|	tok = <RELEASE>
 	|	tok = <RENAME>
+	|   tok = <REPEAT>
```
Comment (MurrayBrown) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj]: nit: whitespace

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java [hunk 2]
```diff
@@ -2322,6 +2322,59 @@ public StringDataValue concatenate(
...
+            }
+
+            if (repeatedTimes < 0) {
+                throw StandardException.newException(
+                        SQLState.LANG_INVALID_FUNCTION_ARGUMENT, rightOperand, "REPEAT");
+            }
+        }
+
+        String val = leftOperand.getString();
+        String concatenatedVal = null;
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java]: Use a StringBuilder here, it's the recommended approach to concatenate several Strings together.

**Comment by dgomezferro** [comment]
Change looks good +1! Minor comment about using StringBuilder

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj (hunk 1 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/sqlgrammar.jj]: fixed.

File: db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java (hunk 2 above)
Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java]: Thanks @dgomezferro . I've pushed a commit to use StringBuilder.

**Pull request merged**